CandidateExercise/ConfigurationExercise/Config.cs

```diff
@@ -88,42 +113,181 @@ public void Shutdown()
 
         public ConfigValue GetValue(string key)
         {
-            throw new NotImplementedException();
+            _log.Info("In GetValue");
+            try
+            {
+                var persistenceStorage = _persistSource;
+                KVPList kvList = GetValues(new List<String>() { key });
+                if (kvList == null)
+                    throw new PersistException("cannon get the value " + key, null);
+                return kvList[0].Value;
+            }
+            catch (PersistException ex)
+            {
+                _log.Info("Persist Exception in  GetValue : " + ex.Message);
+                throw ex;
```

Comment:
You've got a lot of these 2 repeated lines (the log and the throw). Could you create a helper method to condense these lines?